pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -334,47 +325,4 @@ func (Plugin) OnInboundRouteConfiguration(in *plugin.InputParams, route *xdsapi.
 // OnOutboundCluster implements the Plugin interface method.
 func (Plugin) OnOutboundCluster(env model.Environment, node model.Proxy, service *model.Service,
 	servicePort *model.Port, cluster *xdsapi.Cluster) {
```

Comment:
change this method signature to take the DestinationRule as well. Same goes for Inboundcluster. Then you can simply do if destRule != nil and destRule.trafficPolicy.TLSmode==istio_mutual then...
don't forget to extract the port specific traffic policy first.

Reply:
cc @ostromart

Reply:
It seems most TLS related settings are in cluster.go - it may keep them grouped ( and maybe later move to a 'tls.go' plugin). The authentication plugin seems to deal mostly with the auth policies - while cluster with the network destination rules.

Reply:
@rshriram could we proceed as it is right now, given other TLSettings mode is also handled in cluster.go? And later if needed extract client tls stuff out into tls.go plugin, suggested by @costinm 

I'll send follow-up PR to actually build multi-filter chain + tls sniffing filter, which depends on this one to be checked-in.

Thanks!

Reply:
Sure. But can you push this PR to 0.8 as well ? This is independent of any thing else.